dace/transformation/interstate/fpga_transform_state.py

```diff
@@ -185,12 +187,14 @@ def apply(self, sdfg):
                         allow_conflicts=desc.allow_conflicts,
                         strides=desc.strides,
                         offset=desc.offset)
+                    fpga_array[1].location = desc.location
+                    desc.location.clear()
```

Comment:
Are you sure this works? I think `fpga_array[1].location = desc.location` assigns a pointer to the dictionary, so you will still clear it.

Why does it need to be deleted?

Reply:
No it does work, because python automatically duplicates objects pointed to by multiple references if their state diverges. (Tbh I don't fully get the exact rules for this either). It needs to be cleared, because desc will be the array used on the host, i.e. it does not have the same memory properties as the one on fpga.